### ISSUE TITLE
Error opening link URLs in notes such as file://<path>/document.pdf#page=42

### ISSUE THREAD

**Issue by ackalker** [zim-desktop-wiki/zim-desktop-wiki#1412]
## Zim version
```
$ zim -v | head -n 1
zim 0.73.5
```
## Scenario
I'm trying to add links to specific pages in locally stored PDFs to my notes, for example: `file:///home/****/Resources/STM32/docs/dm00237416-stm32f7-series-and-stm32h7-series-cortexm7-processor-programming-manual-stmicroelectronics.pdf#page=42`.
## Expected results
* Zim should convert the entire URL into a clickable link.
* When clicking on the link, Zim should open the program associated with `file://` URLs (which in my case is Chromium) with the PDF document at the requested page.
## Actual results
* Zim converts the entire URL into a clickable link.
* Clicking on the created link produces an error: `No such file or folder: /home/****/Resources/STM32/docs/dm00237416-stm32f7-series-and-stm32h7-series-cortexm7-processor-programming-manual-stmicroelectronics.pdf#page=42`.
* When I right-click on the link in my notes, select `Copy link` from the context menu, then paste it in a terminal shell, the pasted text turns out to be: `~/Resources/STM32/docs/dm00237416-stm32f7-series-and-stm32h7-series-cortexm7-processor-programming-manual-stmicroelectronics.pdf#page=42`, i.e. Zim appears to have converted the `file://` URL into what looks like a regular file path, but it left the `#page=42` URL fragment attached.
* When I open my intended URL with Chromium from a terminal, like:
   ```sh
   $ chromium file://home/****/Resources/STM32/docs/dm00237416-stm32f7-series-and-stm32h7-series-cortexm7-processor-programming-manual-stmicroelectronics.pdf#page=42
   ```
   Chromium opens the PDF at the requested page without problems.

**Comment by jaap-karssenberg** [comment]
Yes, we can try solve this by adding code to cut off the "#anchor" part
when parsing the file uri. However, we can not solve it robustly e.g. when
handing off the URL to "system default" it will still fail in most cases.

The reason is that the "file://" uri specification does not account for
handling of "#anchor" part of the URI. So there is no standard way of
handling them. Strictly speaking what zim does today is correct.

See rfc8089 (ietf.org) <https://datatracker.ietf.org/doc/html/rfc8089> for
the specification

Regards,

Jaap
